src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,198 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int size;
+
     @Override
     public void add(T value) {
+        putNode(value, size == 0 ? "first" : "top");
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size);
+        if (index > 0 && index < size) {
+            putNodeMiddle(value, getNodeByIndex(index));
+            return;
+        }
+        putNode(value, index == 0 ? size == 0 ? "first" : "bottom" : "top");
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null || list.isEmpty()) {
+            throw new NoSuchElementException("You can't add any data from empty list");
+        }
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, size - 1);
+        return getNodeByIndex(index).getElement();
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index, size - 1);
+        Node<T> node = getNodeByIndex(index);
+        T element = node.getElement();
+        node.setElement(value);
+        return element;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index, size - 1);
```

Comment:
```suggestion
        checkIndex(index);
```
why -1? 
it is hard to read, in fact you dont need to pass another value in checkIndex() method